### PR TITLE
Refactor Button

### PR DESCRIPTION
**Pull request by luistorres**
* It adds a link to form
* It puts the button styles in a component, so they'll be easier to 
update soon
* Adds sections tags in Sections components and ids

### CODE REVIEW

File: frontend/components/Button/index.css [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+@import '../../styles/layout.css';
+
+.root {
+  display: flex;
+  box-sizing: border-box;
+  padding: $spacing-12 $spacing-20;
+
+  background-color: $yellow;
+
+  color: inherit;
```
Comment (jfranciscosousa) [inline frontend/components/Button/index.css:12]: Isn't this default behavior?

File: frontend/components/Button/index.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import Text from 'root/components/Text';
+
+import './index.css';
+
+const Button = ({ handleClick, href }) => {
+  const Component = href ? 'a' : 'button';
+
+  return (
+    <Component styleName="root" type="submit" href={href} onClick={handleClick}>
```
Comment (jfranciscosousa) [inline frontend/components/Button/index.jsx]: I don't know... we should not have `href` in a button component and the same thing applies to the `type` in `a` tags, we should make this explicit or remove `type` in `a` tags and `href` in buttons.

File: frontend/components/Button/index.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      <Text fontFamily="mono" color="red" weight="bold">
+        #inscreve-te
+      </Text>
+    </Component>
+  );
+};
+
+Button.defaultProps = {
```
Comment (jfranciscosousa) [inline frontend/components/Button/index.jsx]: `defaultProps` after `propTypes` no? Makes more sense to me I at least.

File: frontend/components/Button/index.jsx (hunk 2 above)
Comment (luistorres) [reply frontend/components/Button/index.jsx]: Buttons and links should have the same styles either, but they do, so I'm treating them as they were the same to make my work easier.

What good would it be to have 2 components to do the exact same thing? Yet, I agree type shouldn't be there

File: frontend/components/Button/index.jsx (hunk 3 above)
Comment (luistorres) [reply frontend/components/Button/index.jsx]: If it makes the Doc happier, ok

File: frontend/components/Button/index.css (hunk 1 above)
Comment (luistorres) [reply frontend/components/Button/index.css:12]: The link has that weird things that change the color of the text. Not sure if I really need this one, but it is just to be sure that it resets them

File: frontend/components/Button/index.jsx (hunk 2 above)
Comment (jfranciscosousa) [reply frontend/components/Button/index.jsx]: It's ok, merge. I think react omits `null` attributes from the HTML output.

**Review by jfranciscosousa — APPROVED** [review]
:statue_of_liberty:

**Pull request merged**